examples/with-nextjs/src/app/components/ImpersonationShield.tsx

```diff
@@ -38,7 +25,12 @@ export function ImpersonationShield({ data }: ImpersonationShieldProps) {
 
   const { display_name, nameguard_result } = data;
 
-  const textClass = cc(["font-semibold", textColor(nameguard_result?.rating)]);
+  if (!nameguard_result) return <LoadingShield />;
+
+  const textClass = cc([
+    "font-semibold",
+    getNameGuardRatingTextColors(nameguard_result.rating),
+  ]);
 
   const Icon = nameguard_result?.rating === Rating.warn ? WarnIcon : AlertIcon;
```

Comment:
Why do we have the ? after `nameguard_result` here? Same idea for the rest of the code here in this function.